File: iosApp/iosApp/Presentation/Views/FilterOptions/SortType.swift
//
//  FilterKeyword.swift
//  iosApp
//
//  Created by rescalon on 10/9/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation

enum SortType: CaseIterable, Identifiable {
    case FirstAdded
    case LastAdded
    
    var displayName: String {
        switch self {
        case .FirstAdded: "First Added"
        case .LastAdded: "Last Added"
        }
    }
    
    var sortType: String {
        switch self {
        case .FirstAdded: "created_at.asc"
        case .LastAdded: "created_at.desc"
        }
    }
    
    var id: String { return self.displayName }
}

File: androidApp/src/main/java/com/escalondev/movies_app_kmp/android/ui/filter/SortType.kt
package com.escalondev.movies_app_kmp.android.ui.filter

enum class SortType(val displayName: String, val sortType: String) {
    FIRST_ADDED("First Added", "created_at.asc"),
    LAST_ADDED("Last Added", "created_at.desc")
}
